src/enclave/enclave_oe.c

```diff
@@ -351,8 +374,12 @@ static void _copy_shared_memory(const sgxlkl_shared_memory_t* host)
         for (size_t i = 0; i < enc->num_virtio_blk_dev; i++)
         {
             enc->virtio_blk_dev_mem[i] = host->virtio_blk_dev_mem[i];
+            sgxlkl_ensure_outside(
+                enc->virtio_blk_dev_mem[i], sizeof(struct virtio_dev));
+
             const char* name = host->virtio_blk_dev_names[i];
             size_t name_len = oe_strlen(name) + 1;
```

Comment:
If the name string is manipulated by the malicios host (for exmaple, no NULL terminator), `strlen()` might read memory from both untrusted memory and enclave memory, exposing side channel info of enclave memory (whether a byte is NULL or not)

Reply:
Yes, that's on my to-do list.